src/SourceWatcherFuncApp/DiagSourceWatcher.cs

```diff
@@ -0,0 +1,87 @@
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Linq;
+using SourceWatcherFuncApp.Services;
+using SourceWatcherFuncApp.Utilities;
+using SourceWatcherFuncApp.Entities;
+namespace Diag.SourceWatcher
+{
+    public class DiagSourceWatcher
+    {        
+        private IGithubService githubService;
+
+        private ITableStorageService storageService;
+
+        private IConfigurationRoot config;
+
+        private IBlobService blobService;
+
+        public DiagSourceWatcher(IConfigurationRoot configurationRoot, IGithubService GithubService, 
+                                 ITableStorageService StorageService, IBlobService BlobService)
+        {
+            githubService = GithubService;
+            config = configurationRoot;
+            storageService = StorageService;
+            blobService = BlobService;
+        }
+
+        [FunctionName("DiagSourceWatcher")]
+        public async Task Run([TimerTrigger("0 0 0 * * *")]TimerInfo timerInfo, ILogger log)
+        {
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+
+            log.LogInformation($"Loading directories from github branch  {config["Github:Branch"]}");
+
+            var githubDirectories = await githubService.DownloadGithubDirectories(config["Github:Branch"]);
```

Comment:
Check `Github:Branch` for null or pass `branchdownloadUrl`

Reply:
`Github:Branch` is checked for null within the method